pdns/dnswriter.cc

```diff
@@ -236,25 +236,30 @@ void DNSPacketWriter::xfrLabel(const string& Label, bool compress)
       d_labelmap.push_back(make_pair(chopped, pos));                       //  if untrue, we need to count - also, don't store offsets > 16384, won't work
     }
 
+    if(label.size() > 255)
```

Comment:
This ignores that the name may be escaped (see the if below). How about comparing d_record.size before and after?